jams/core.py

```diff
@@ -760,15 +760,18 @@ def validate(self, strict=True):
         JObject.validate
         '''
 
-        valid = super(Annotation, self).validate(strict=strict)
-
         # Get the schema for this annotation
-        ann_schema = schema.namespace(self.namespace)
+        ann_schema = schema.namespace_array(self.namespace)
+
+        valid = True
 
         try:
+            jsonschema.validate(self.__json_light__(data=False),
+                                schema.JAMS_SCHEMA)
+
             # validate each record in the frame
-            for rec in self.data:
-                jsonschema.validate(serialize_obj(rec), ann_schema)
+            data_ser = [serialize_obj(obs) for obs in self.data]
```

Comment:
Is this where the speed up is coming from?

Reply:
That's the first speedup.  The second speedup comes from eliminating redundant serialization calls, as noted above in-thread.